File: LeetCode/week-249/1930.长度为3的不同回文子序列.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        ans=0
        for i in range(26):
            L=0
            R=len(s)-1
            while L<len(s) and s[L]!=chr(i+97):L+=1
            while R>0 and s[R]!=chr(i+97):R-=1
            if L>=R:continue
            res=[0]*26
            for j in range(L+1,R):
                res[ord(s[j])-97]=1
            for x in res:ans+=x
        return ans

File: LeetCode/week-249/1930.长度为3的不同回文子序列.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        int ans=0;
        for(int i=0;i<26;++i){
            int L=0, R=s.size()-1;
            while(L<s.size() and s[L]!=i+'a')L+=1;
            while(R>0 and s[R]!=i+'a')R-=1;
            if(L>=R)continue;
            vector<int> res(26);
            for(int j=L+1;j<R;j++)res[s[j]-'a']=1;
            for(auto x : res)ans+=x;
        }
        return ans;
    }
};